docs/dataset/advanced-usage/set-up-natural-language-search.md

```diff
@@ -0,0 +1,157 @@
+
+## Example
+
+The [`kolena`](https://github.com/kolenaIO/kolena) repository contains a runnable
+[example](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/search_embeddings) for embeddings extraction and
+upload. This builds off the data uploaded in the
+[semantic_segmentation](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/semantic_segmentation)
+example dataset, and is best run after this data has been uploaded to your Kolena environment.
+
+## How to Set Up Natural Language Search
+
+Uploading embeddings to Kolena can be done in three simple steps:
+
+- [**Step 1**](#step-1-install-kolena_embeddings-package): installing dependency package
+- [**Step 2**](#step-2-load-dataset-and-model): loading dataset and model to run embedding extraction
+- [**Step 3**](#step-3-load-images-for-extraction): loading images for input to extraction library
+- [**Step 4**](#step-4-extract-and-upload-embeddings): extracting and uploading search embeddings
+
+Let's take a look at each step with example code snippets.
+
+### Step 1: Install `kolena_embeddings` Package
+
+The package can be installed via `pip` or `poetry` and requires use of your kolena token which can be created
+on the [:kolena-developer-16: Developer](https://app.kolena.io/redirect/developer) page.
+
+We first [retrieve and set](../../installing-kolena.md#initialization) our `KOLENA_TOKEN` environment variable.
+This is used by the uploader for authentication against your Kolena instance.
+
+```shell
+export KOLENA_TOKEN="********"
+```
+
+=== "`pip`"
+
+    Run the following command, making sure to replace <KOLENA_TOKEN> with the token retrieved from the developer page:
+    ```shell
+    pip install --extra-index-url="https://<KOLENA_TOKEN>@gateway.kolena.cloud/repositories" kolena-embeddings
+    ```
+
+=== "`poetry`"
+
+    Configure an additional poetry source:
+    ```shell
+    poetry source add --priority=supplemental kolena-embeddings "https://gateway.kolena.cloud/repositories"
+    ```
+
+    Then run the following command, making sure to replace <KOLENA_TOKEN> with the token retrieved from the developer page:
+    ```shell
+    poetry config http-basic.kolena-embeddings <KOLENA_TOKEN> ""
+    ```
+
+This package provides the `kembed.util.extract_embeddings` method that generates
+embeddings as a numpy array for a given [`PIL.Image.Image`](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image)
+object.
+
+### Step 2: Load Dataset and Model
+
+Before extracting embeddings on a dataset, we need to load the dataset. The dataset
+seeded in the [semantic_segmentation](https://github.com/kolenaIO/kolena/tree/trunk/examples/dataset/semantic_segmentation)
+example contains [`Image`][kolena.asset.ImageAsset] assets referenced by the `locator`
+column, and we load the dataset in to a dataframe.
+
+The embedding model and its key are obtained via the `load_embedding_model()` method.
+
+```python
```

Comment:
Also include `kolena.initialize(verbose=True)` in this script? copy+paste and running leads to:

```
kolena.errors.UninitializedError: client has not been initialized via kolena.initialize(...)
```